File: exercicios_java/exercicios_java/src/secao7/exercicio7.java
/*
Primeira tarefa testar 200 mouses, anotando estado deles. O programa deve receber
um número indeterminado de entradas, contendo identificação do mouse, podendo ser:
Necessita de esfera, limpeza, troca de cabo/conector, quebrado

Identificação = 0 encerra. O programa deve emitir o relatório:
1 - necessita da esfera		
2 - de limpeza\n3 - troca do cabo ou conector\n4 - quebrado

quantidade = 0
qtd_esf = 0
qtd_limp = 0
qtd_tcc = 0
qtd_qb = 0
*/
package secao7;
import java.util.Scanner;

public class exercicio7{
    public static void main(String[] args){
        int ident = -1, quant = -1, qesf = 0, qlimp = 0, qtcc = 0, qqb = 0;
        Scanner t = new Scanner(System.in);

        while(ident != 0){
            System.out.print("0 encerra\n1 - necessita da esfera\n2 - de limpeza\n3 - troca do cabo ou conector\n4 - quebrado\n");
            System.out.print("Digite o identificador: ");
            ident = t.nextInt();
            quant += 1;
            if(ident == 1){
                qesf += 1;
            }else if(ident == 2){
                qlimp += 1;
            }else if(ident == 3){
                qtcc += 1;
            }else if(ident == 4){
                qqb += 1;
            }
        }float i1, i2, i3, i4;
        i1 = ((float)qesf / (float)quant * (float)100.00);
        i2 = ((float)qlimp / (float)quant * (float)100.00);
        i3 = ((float)qtcc / (float)quant * (float)100.00);
        i4 = ((float)qqb / (float)quant * (float)100.00);

        System.out.print("Quantidade: " + quant);
        System.out.print("\nId | Mouses | Pct |");
        System.out.printf("\n1 | %d | %.2f |", qesf, i1);
        System.out.printf("\n2 | %d | %.2f |", qlimp, i2);
        System.out.printf("\n3 | %d | %.2f |", qtcc, i3);
        System.out.printf("\n4 | %d | %.2f |", qqb, i4);

        t.close();
    }
}

File: exercicios_c/secao7/exercicio7.c
//Primeira tarefa testar 200 mouses, anotando estado deles. O programa deve receber
//um número indeterminado de entradas, contendo identificação do mouse, podendo ser:
//Necessita de esfera, limpeza, troca de cabo/conector, quebrado

//Identificação = 0 encerra. O programa deve emitir o relatório:
//1 - necessita da esfera		
//2 - de limpeza
//3 - troca do cabo ou conector
//4 - quebrado

#include <stdio.h>

int main(){
    int qtd_esf = 0, qtd_lp = 0, qtd_tcc = 0, qtd_qb = 0, iden;
    float pqesf, pqlp, pqtcc, pqqb, quantidade = 0.0;
    
    printf("Digite o identificador: ");
    scanf("%d", &iden);

    while (iden != 0){
        if (iden == 1){
            qtd_esf += 1;
        }if (iden == 2){
            qtd_lp += 1;
        }if (iden == 3){
            qtd_tcc += 1;
        }if (iden == 4){
            qtd_qb += 1;
        };
        quantidade += 1;
        printf("1 - necessita da esfera\n2 - de limpeza\n3 - troca do cabo ou conector\n4 - quebrado\n");
        printf("Digite o identificador: ");
        scanf("%d", &iden);        
    }
    pqesf = qtd_esf/quantidade*100;
    pqlp  = qtd_lp/quantidade*100;
    pqtcc  = qtd_tcc/quantidade*100;
    pqqb  = qtd_qb/quantidade*100;
    printf("|   Situacao    |   Quantidade  |   Percentual    |\n");
    printf("------------------------------------------------------");
    printf("|      1        |       %d      |      %.1f%%      |\n", qtd_esf, pqesf);
    printf("|      2        |       %d      |      %.1f%%      |\n", qtd_lp, pqlp);
    printf("|      3        |       %d      |      %.1f%%      |\n", qtd_tcc, pqtcc);
    printf("|      4        |       %d      |      %.1f%%      |", qtd_qb, pqqb);
    printf("------------------------------------------------------");
}